csrc/executor.cpp

```diff
@@ -1857,7 +1863,24 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
 
   if (execute_kernel_) {
     ensureAvailableDynamicSmemSize(executor_entry->launch_params.smem());
-    auto arg_buffer = args.getBuffer(kernel()->indexType());
+    std::vector<TensorView*> tvs;
+    for (auto val : kernel()->inputs()) {
+      tvs.emplace_back(dynamic_cast<TensorView*>(val));
+    }
+    for (auto val : kernel()->outputs()) {
+      tvs.emplace_back(dynamic_cast<TensorView*>(val));
+    }
+    for (auto alloc : kernel()->summary().global_allocations) {
+      auto tv = alloc->buffer()->as<TensorView>();
+      if (tv->isFusionOutput()) {
+        continue;
+      }
+      tvs.emplace_back(tv);
+    }
+    if (lowered_->kernel()->summary().max_rng_offsets >= 0) {
+      tvs.emplace_back(nullptr);
+    }
```

Comment:
nit: can you refactor this as a function. Executor functions tend to grow quickly unless intentionally make it succinct.

Reply:
refactored as `getTvsForKernelArguments`

Reply:
reFactored as `getTvsForKernelArguments`

Reply:
refactored as `getTvsForKernelArguments`

Reply:
reFactored as `getTvsForKernelArguments`